pkg/domain/infra/abi/containers.go

```diff
@@ -264,30 +264,30 @@ func (ic *ContainerEngine) ContainerRm(ctx context.Context, namesOrIds []string,
 	reports := []*entities.RmReport{}
 
 	names := namesOrIds
-	for _, cidFile := range options.CIDFiles {
-		content, err := ioutil.ReadFile(cidFile)
-		if err != nil {
-			return nil, errors.Wrap(err, "error reading CIDFile")
-		}
-		id := strings.Split(string(content), "\n")[0]
-		names = append(names, id)
-	}
-
 	// Attempt to remove named containers directly from storage, if container is defined in libpod
 	// this will fail and code will fall through to removing the container from libpod.`
 	tmpNames := []string{}
 	for _, ctr := range names {
 		report := entities.RmReport{Id: ctr}
-		if err := ic.Libpod.RemoveStorageContainer(ctr, options.Force); err != nil {
+		report.Err = ic.Libpod.RemoveStorageContainer(ctr, options.Force)
+		switch errors.Cause(report.Err) {
+		case nil:
 			// remove container names that we successfully deleted
-			tmpNames = append(tmpNames, ctr)
-		} else {
 			reports = append(reports, &report)
+		case define.ErrNoSuchCtr:
+			// There is still a potential this is a libpod container
+			tmpNames = append(tmpNames, ctr)
```

Comment:
I think we duplicate ctr in tmpNames here - we go through the switch and then append again after the switch

Reply:
No the case statement does not fall though to the default block.

Reply:
It doesn't need to fall through - the append happens again after the switch block ends.

Reply:
Could you send a link to the exact line, you are talking about? Maybe I am missing it, but this is the default block, and there is no append after this.
https://github.com/containers/podman/blob/master/pkg/domain/infra/abi/containers.go#L280-L288

Reply:
Nevermind, you're right, I'm misreading it